static/src/javascripts/projects/commercial/modules/liveblog-adverts.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+import detect from 'lib/detect';
+import mediator from 'lib/mediator';
+import { addSlot } from 'commercial/modules/dfp/add-slot';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import createSlot from 'commercial/modules/dfp/create-slot';
+import spaceFiller from 'common/modules/article/space-filler';
+
+const INTERVAL = 5; // number of posts between ads
+const OFFSET = 1.5; // ratio of the screen height from which ads are loaded
+const MAX_ADS = 8; // maximum number of ads to display
+
+type spaceFillerRules = {
+    bodySelector: string,
+    slotSelector: string,
+    fromBottom: boolean,
+    startAt: Node | null | typeof undefined, // #? I feel this is circumventing the point of Flow
+    absoluteMinAbove: number,
+    minAbove: number,
+    minBelow: number,
+    filter: (slot: { top: number }, index: number) => boolean,
+};
+
+let SLOTCOUNTER = 0;
+let WINDOWHEIGHT;
+let firstSlot;
+
+const startListening = () => {
+    // eslint-disable-next-line no-use-before-define
+    mediator.on('modules:autoupdate:updates', onUpdate);
+};
+
+const stopListening = () => {
+    // eslint-disable-next-line no-use-before-define
+    mediator.off('modules:autoupdate:updates', onUpdate);
+};
+
+const getWindowHeight = (doc = document): number => {
+    if (doc.documentElement && doc.documentElement.clientHeight) {
+        return doc.documentElement.clientHeight;
+    }
+    return 0; // #? zero, or throw an error?
```

Comment:
Smart 😄  I was just wondering why you have to test for `documentElement` and `clientHeight`. They don't appear to be nullable (See [#1](https://www.w3.org/TR/DOM-Level-2-Core/core.html#ID-87CD092) and [#2](https://drafts.csswg.org/cssom-view/#dom-element-clientheight)).